### ISSUE TITLE
CentOS 7 - CoTURN v4.5.0.6 wont run

### ISSUE THREAD

**Issue by shamun** [coturn/coturn#118]
CentOS 7, Linux localhost.localdomain 3.10.0-514.6.2.el7.x86_64, CoTURN v4.5.0.6

**Step 1**  - Download & Install http://turnserver.open-sys.org/downloads/v4.5.0.6/
**Step 2** - Run/Start

`systemctl restart turnserver.service`

**Step 3** -  FAIL

```
[root@localhost ~]# systemctl restart turnserver.service
Job for turnserver.service failed because a configured resource limit was exceeded. See "systemctl status turnserver.service" and "journalctl -xe" for details.
[root@localhost ~]# systemctl status turnserver.service
● turnserver.service - coturn
   Loaded: loaded (/usr/lib/systemd/system/turnserver.service; disabled; vendor preset: disabled)
   Active: failed (Result: resources) since Wed 2017-04-05 08:50:26 CEST; 15s ago
     Docs: man:coturn(1)
           man:turnadmin(1)
           man:turnserver(1)
  Process: 2460 ExecStart=/usr/bin/turnserver -o -c /etc/turnserver/turnserver.conf $EXTRA_OPTIONS (code=exited, status=0/SUCCESS)

Apr 05 08:50:25 localhost.localdomain turnserver[2465]: 0: Trying to bind fd 18 to <127.0.0.1:80>: errno=13
Apr 05 08:50:25 localhost.localdomain turnserver[2465]: 0: Cannot bind TLS/TCP listener socket to addr 127.0.0.1:80
Apr 05 08:50:25 localhost.localdomain turnserver[2465]: 0: Trying to bind TLS/TCP listener socket to addr 127.0.0.1:80, again...
Apr 05 08:50:26 localhost.localdomain turnserver[2465]: 0: Trying to bind fd 16 to <127.0.0.1:80>: errno=13
Apr 05 08:50:26 localhost.localdomain turnserver[2465]: 0: Cannot bind DTLS/UDP listener socket to addr 127.0.0.1:80
Apr 05 08:50:26 localhost.localdomain turnserver[2465]: 0: Fatal final failure: cannot bind DTLS/UDP listener socket to addr 127.0.0.1:80
Apr 05 08:50:26 localhost.localdomain systemd[1]: turnserver.service never wrote its PID file. Failing.
Apr 05 08:50:26 localhost.localdomain systemd[1]: Failed to start coturn.
Apr 05 08:50:26 localhost.localdomain systemd[1]: Unit turnserver.service entered failed state.
Apr 05 08:50:26 localhost.localdomain systemd[1]: turnserver.service failed.
```

**Manual fix:**
=

```
[Unit]
Description=rfc5766-turn-server
Documentation=man:rfc5766-turn-server(1) man:turnadmin(1) man:turnserver(1)
After=syslog.target network.target

[Service]
Type=forking
EnvironmentFile=/etc/sysconfig/turnserver
PIDFile=/var/run/turnserver.pid
ExecStart=/usr/bin/turnserver -o -c /etc/turnserver/turnserver.conf $EXTRA_OPTIONS
ExecStopPost=/usr/bin/rm -f /var/run/turnserver.pid
Restart=on-abort

[Install]
WantedBy=multi-user.target
```

**Comment by misi** [comment]
@shamun 
Could you point out the exact fix?
I guess you can't run ports under 24 without cap or root right.
I don't see any error here. Could you please help me understand your issue?

**Comment by guusdk** [comment]
@misi: I'm guessing that the 'manual fix' is removing the User and Group definition that's typically in the server file. That might cause it to be executed as root, preventing the need something like `setcap cap_net_bind_service,cap_net_admin+pe /usr/sbin/sslh` if you want to use port numbers under 1024.

**Comment by misi** [comment]
@guusdk coTURN drops priv after binding port so AFAIU it could run even under ports 1024

**Comment by misi** [comment]
Is there still any open question or could we close this issue?

**Comment by rapsys** [comment]
The issue is not closed.

It seems that turnserver don't send the notification required by Type=notify :
https://www.freedesktop.org/software/systemd/man/systemd.service.html
> **it is expected that the service sends a notification message via sd_notify(3) or an equivalent call when it has finished starting up.**

How do we turn on this notification ?

An other way is to revert the turnserver.service to something like that :
```
[Unit]
Description=coturn
Documentation=man:coturn(1) man:turnadmin(1) man:turnserver(1)
After=syslog.target network.target

[Service]
User=turnserver
Group=turnserver
Type=exec
PIDFile=/run/turnserver/turnserver.pid
EnvironmentFile=/etc/sysconfig/turnserver
ExecStart=/usr/bin/turnserver -c /etc/turnserver/turnserver.conf $EXTRA_OPTIONS
ExecStopPost=/usr/bin/rm -f /run/turnserver/turnserver.pid
Restart=on-abort

LimitCORE=infinity
LimitNOFILE=999999
LimitNPROC=60000
LimitRTPRIO=infinity
LimitRTTIME=7000000
CPUSchedulingPolicy=other
UMask=0007

[Install]
WantedBy=multi-user.target
```

**Comment by rapsys** [comment]
Affected version is 4.5.2

To reproduce, configure your coturn, then :
`systemctl start turnserver.service`

It will hang forever and timeout after a while...

**Comment by eakraly** [comment]
@rapsys did you find a solution?
Could you please submit a PR with a proposed fix?

**Comment by rapsys** [comment]
Pull request done here, better late than never : https://github.com/coturn/coturn/pull/990

**Comment by eakraly** [comment]
Fixed by #1043
